### PR TITLE
[WIP] Reworking Python build to support pip deployment.

### PR DESCRIPTION
**Pull request by metasim**
Aims to integrate Python `setuptools` build with sbt to create a binary artifact sufficient for pyspark deployment. Removes dependency on `sbt-spark-packages`, which is languishing.

## Misc Resources

* https://setuptools.readthedocs.io/en/latest/setuptools.html#id4
* https://github.com/pypa/sampleproject
* https://docs.python.org/3.7/distutils/setupscript.html
* https://docs.python.org/2/distutils/setupscript.html#distutils-additional-files
* https://www.python.org/dev/peps/pep-0427/
* https://jichu4n.com/posts/how-to-add-custom-build-steps-and-commands-to-setuppy/
* https://github.com/anguenot/pyspark-cassandra
* https://bytes.grubhub.com/managing-dependencies-and-artifacts-in-pyspark-7641aa89ddb7
* https://github.com/apache/spark/blob/master/python/setup.py
* https://github.com/locationtech/geomesa/blob/master/geomesa-spark/geomesa_pyspark/pom.xml

### CODE REVIEW

File: project/PIPBuildPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        val java = (Compile / packageBin / artifact).value
+        java.withType("zip").withClassifier(Some("python")).withExtension("zip")
+      },
+      packageBin / artifactPath := {
+        val art = (packageBin / artifact).value
+        val dest = (Compile / target).value
+        val ver = version.value
+        dest / s"${art.name}-python-$ver.zip"
+      },
+      test := {
```
Comment (metasim) [inline project/PIPBuildPlugin.scala:112]: This doesn't yet work. Needs to set up the pyspark jar paths and test directories.

**Review by bguseman — COMMENTED** [review]
Without fully grokking all the details, this looks solid. Will a pip install pull in everything that's needed including jvm stuff?

**Comment by metasim** [comment]
Not sure about pulling in the jvm stuff. I'm wondering if something in the `__init__.py` might be required.

**Comment by metasim** [comment]
Replaced by https://github.com/locationtech/rasterframes/pull/125

**Pull request closed without merging**